src/main/java/pro/trevor/tankgame/rule/impl/version3/range/DeadTankRange.java

```diff
@@ -0,0 +1,13 @@
+package pro.trevor.tankgame.rule.impl.version3.range;
+
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.state.range.FunctionVariableRange;
+
+import java.util.stream.Collectors;
+
+public class DeadTankRange<S> extends FunctionVariableRange<S, Tank> {
+    public DeadTankRange(String name) {
+        super(name, (state, subject) -> state.getBoard().gatherUnits(Tank.class)
```

Comment:
Seems like the difference between this and LivingTankRange is the filter function.  Seems like this could a more generic class for collecting units matching certain conditions (maybe something like `new UnitRange("living tanks", Tank.class, Tank::isDead)`)

Reply:
Working on this